### ISSUE TITLE
False positive "Call to untyped function in typed context" error

### ISSUE THREAD

**Issue by playpauseandstop** [python/mypy#2949]
Hi,

I'm using `mypy` for a while now, but after update to `0.501` + `typeshed==1.0.1` start getting strange `Call to untyped function "Table" in typed context` errors. Below some technical info about errors,

`mypy.ini`,

```ini
[mypy]
check_untyped_defs = True
disallow_untyped_calls = True
disallow_untyped_defs = True
follow_imports = skip
ignore_missing_imports = True
strict_optional = True
warn_redundant_casts = True
warn_unused_ignores = True
```

`app/pseudocode.py`,

```py
import sqlalchemy as sa

from app.models import metadata

dummy = sa.Table(
    'table', metadata,

     sa.Column('id', sa.Integer(), primary_key=True),
)
```

```
$ vagga _run container mypy app/pseudocode.py
app/pseudocode.py:5: error: Call to untyped function "Table" in typed context
```

----

But from my perspective this is wrong mypy behaviour as,

1. First of all, `sa.Table` is not a function (it's more about proper error message)
2. It's global scope. Is it possible to disable type checking in global scope? Or if I have `disallow_untyped_calls` enabled it will check even global scope then? Why that was not a case for `mypy==0.471` for example?
3. `sa.Table` has type annotation in [python/typeshed](https://github.com/python/typeshed/blob/master/third_party/2and3/sqlalchemy/__init__.pyi#L106) and I for sure has latest version of typeshed installed in my container,

```
$ vagga _run container cat /usr/local/lib/mypy/typeshed/third_party/2and3/sqlalchemy/__init__.pyi | grep -n Table
106:    Table,
```

----

With that in mind, I'm now ignoring all those false positives with `# type: ignore`, but hopefully this behaviour will fixed soon.

**Comment by gvanrossum** [comment]
The root cause is that there are no annotations for the `Table` constructor here:
https://github.com/python/typeshed/blob/master/third_party/2and3/sqlalchemy/sql/schema.pyi#L23

In fact the mysql stubs in general have no argument annotations. This would be a huge undertaking to fix (given the size and subtlety of the API).

**Comment by playpauseandstop** [comment]
@gvanrossum 

Only question why untyped calls in global scope became an error behaviour in `0.501`? I mean, I expect that `disallow_untyped_calls` errors only raised by,

```py
def func_one(number):
    return number * 2

def func_two(number: int) -> str:
    return str(func_one(number))
```

But not when call made in global scope,

```py
if __name__ == '__main__':
    print(func_one(int(sys.argv[1])))
```

----

In same time I understand now the problem with missing type hints for `sqlalchemy.Table` constructor, that was my mistake to pretend that `Table` had typed context, when it doesn't :(

**Comment by gvanrossum** [comment]
Untyped calls at the global scope were always disallowed with that flag. Maybe something else changed, e.g. the sqlalchemy stubs?

**Comment by playpauseandstop** [comment]
My issue fixed after update `mypy` to `0.511`. Seems like `typeshed` has proper SQLAlchemy stubs now. 

Closing this as `fixed`
